### ISSUE TITLE
[no-inline-styles] ternary style gets mistakenly reported

### ISSUE THREAD

**Issue by uloco** [Intellicode/eslint-plugin-react-native#317]
I get a warning on this inline style:

```tsx
<View
  style={[
    styles.safeAreaTransparent,
    { height: props.insetsTop ? insets.top : 0, },  // <---- eslint(react-native/no-inline-styles)
  ]}
/>
```

Does this behaviour work as expected?
If so, how should we change these?

I think it should not report as a warning though ...

**Comment by nobi1007** [comment]
Yeah, all conditional expressions by default are considered invalid [here](https://github.com/Intellicode/eslint-plugin-react-native/blob/4b7149ea7019d288d6318dfabb05461f3134a975/lib/util/stylesheet.js#L405-L409). I think plugin should accept some pattern config to apply only certain validations and skip others.

**Comment by uloco** [comment]
I think it should be configureable, so that if there are variables in it, it should not report at all. Because it makes the code much more complicated with little benefit. If I am already using an inline style because of a conditional, I don't want to goo to the style definition and look up what style will be applied in the else case for example. But it totally makes sense for disabling  inline styles in general for other cases

**Comment by ridvanaltun** [comment]
I think the plugin should accept conditional expressions etc.
